File: python/calculate.py
def calculate_statistic(data):
    """
    Calculates a statistic, e.g., the average, from the data. Assumes data is not empty.
    Corner cases:
    - If the data is empty, the function throws DivisionByZeroError.
    """
    return sum(data) / len(data)

File: typescript/src/calculate.ts
export function calculateStatistic(data: any[]) {
  // Calculates the average of the data
  if (data.length === 0) {
    throw new Error('Division by zero error - data is empty');
  }
  const sum = data.reduce((acc, val) => acc + val, 0);
  return sum / data.length;
}
